test/e2e/suite/issuers/selfsigned/certificaterequest.go

```diff
@@ -125,6 +125,69 @@ var _ = framework.CertManagerDescribe("SelfSigned CertificateRequest", func() {
 			Expect(err).NotTo(HaveOccurred())
 		})
 
+		It("should be able to obtain an Ed25519 Certificate backed by a RSA key", func() {
+			// Replace previous key secret with the RSA one
+			_, err := f.KubeClientSet.CoreV1().Secrets(f.Namespace.Name).Update(context.TODO(), newPrivateKeySecret(
+				certificateRequestSecretName, f.Namespace.Name, rootRSAKey), metav1.UpdateOptions{})
+			Expect(err).NotTo(HaveOccurred())
+
+			crClient := f.CertManagerClientSet.CertmanagerV1().CertificateRequests(f.Namespace.Name)
+			By("Creating a CertificateRequest")
+			csr, err := generateEd25519CSR()
+			Expect(err).NotTo(HaveOccurred())
+
+			_, err = crClient.Create(context.TODO(), gen.CertificateRequestFrom(basicCR,
+				gen.SetCertificateRequestCSR(csr),
+			), metav1.CreateOptions{})
+			Expect(err).NotTo(HaveOccurred())
+
+			By("Verifying the Certificate is valid")
+			err = h.WaitCertificateRequestIssuedValid(f.Namespace.Name, certificateRequestName, time.Second*30, rootRSAKeySigner)
+			Expect(err).NotTo(HaveOccurred())
+		})
+
+		It("should be able to obtain an Ed25519 Certificate backed by a ECSDA key", func() {
+			// Replace previous key secret with ECSDA one
+			_, err := f.KubeClientSet.CoreV1().Secrets(f.Namespace.Name).Update(context.TODO(), newPrivateKeySecret(
+				certificateRequestSecretName, f.Namespace.Name, rootECKey), metav1.UpdateOptions{})
+			Expect(err).NotTo(HaveOccurred())
+
+			crClient := f.CertManagerClientSet.CertmanagerV1().CertificateRequests(f.Namespace.Name)
+			By("Creating a CertificateRequest")
+			csr, err := generateEd25519CSR()
+			Expect(err).NotTo(HaveOccurred())
+
+			_, err = crClient.Create(context.TODO(), gen.CertificateRequestFrom(basicCR,
+				gen.SetCertificateRequestCSR(csr),
+			), metav1.CreateOptions{})
+			Expect(err).NotTo(HaveOccurred())
+
+			By("Verifying the Certificate is valid")
+			err = h.WaitCertificateRequestIssuedValid(f.Namespace.Name, certificateRequestName, time.Second*30, rootECKeySigner)
+			Expect(err).NotTo(HaveOccurred())
+		})
```

Comment:
suggestion: these are the tests that are failing; I don't think they're required. The only one we need here is the one below - `"should be able to obtain an Ed25519 Certificate backed by a Ed25519 key"`. Because this certificate is self-signed, the key type which signs the cert is the same as the key type used in the cert (it's the same key).

So, the `CSR` should be signed with the same key we want to use for the eventual certificate.

There _is_ value in a test which looks similar to this - an Ed25519 CSR which is then signed by an RSA or ECDSA issuer - but that wouldn't live in this file, because that would be testing a different kind of issuer (e.g. the CA issuer).

Reply:
noted started writing the test for the ca issuer and noticed it's a bit more complex :P 

Since i needed to create the cert and key + add a function to return the kubernetes api object for it's secret.

Then before each test create a whole new issuer using that new ed25519 secret. 
And clean it up after.

I'll put it on a branch for review separately